detect_secrets/core/audit.py

```diff
@@ -707,45 +701,29 @@ def _get_secret_with_context(
 
 def get_raw_secret_value(
     secret,
-    plugins_used,
-    custom_plugin_paths,
     file_handle,
-    filename,
 ):
     """
     :type secret: dict
     :param secret: see caller's docstring
 
-    :type plugins_used: list
-    :param plugins_used: output of "plugins_used" in baseline. e.g.
-        >>> [
-        ...     {
-        ...         'name': 'Base64HighEntropyString',
-        ...         'base64_limit': 4.5,
-        ...     },
-        ... ]
-
-    :type custom_plugin_paths: Tuple[str]
-    :param custom_plugin_paths: possibly empty tuple of paths that have custom plugins.
-
     :type file_handle: file object
     :param file_handle: Open handle to file where the secret is
 
-    :type filename: str
-    :param filename: this is needed, because PotentialSecret uses this
-        as a means of comparing whether two secrets are equal.
     """
-    plugin = initialize.from_secret_type(
-        secret_type=secret['type'],
-        plugins_used=plugins_used,
-        custom_plugin_paths=custom_plugin_paths,
-    )
+    secret_length = secret['secret_length']
+    line_position = secret['line_position']
+    line_number = secret['line_number']
+
+    file_lines = file_handle.readlines()
+    if len(file_lines) >= line_number:
+        line = file_lines[line_number - 1]
+        possible_secret = line[line_position:line_position + secret_length]
 
-    plugin_secrets = plugin.analyze(file_handle, filename)
+        possible_secret_hash = PotentialSecret.hash_secret(possible_secret)
+        lowered_possible_secret_hash = PotentialSecret.hash_secret(possible_secret.lower())
```

Comment:
Can you share a test case where `.lower()` vs. not makes a difference?

Reply:
I found two examples in which the hashed secret was the hash of the lowercased secret:
Secret Line: "Password": "SomeKindOfPassword123!@#",
The hashed secret is d1b7e1c27fc85c2bb6e5f2923451bbc76b156e72 (SHA1 of 'your_secret') instead of e383adc92711d348f6df827db2452eddcbd32384

Line: secret = YOUR_SECRET
The hashed secret is 94bc66069e1b40e1294da9bc98187564db71eefd (SHA1 of 'somekindofpassword123!@#') instead of 9c423450db31f2428d1f9eea2c152a4f76b21dbd

Reply:
As I'm seeing this mostly with the Secret Keyword plugin, I'm guessing it has to do with this line: https://github.com/Yelp/detect-secrets/blob/master/detect_secrets/plugins/keyword.py#L340

Reply:
Great catch!